app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,43 @@
+class RecipesController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @recipes = Recipe.where(user_id: current_user.id)
+  end
+
+  def show
+    if current_user != Recipe.find(params[:id]).user
+      flash[:notice] = 'You are not authorized to access this page'
+      redirect_to recipes_path
+    end
+    @recipe = Recipe.find(params[:id])
+  end
+
+  def new
+    @recipe = Recipe.new
+  end
+
+  def create
+    @recipe = Recipe.new(recipe_params)
+    @recipe.user_id = current_user.id
+    if @recipe.save
+      redirect_to recipes_path
+    else
+      render 'new'
+    end
+  end
```

Comment:
### Optional
You don't need to mention the user id. Since we have association, we can simply say `@recipe.user = current_user`. It becomes much cleaner and easier to read as well. Rails will take care of using relevant column in user table for the reference.